src/CovidAppointmentTable.js

```diff
@@ -65,22 +87,51 @@ export default function CovidAppointmentTable() {
 	const [onlyShowAvailable, setOnlyShowAvailable] = useState(true);
 
 	useEffect(() => {
+		resetData()
+	}, []);
+
+	useEffect(()=> {
+		setLoading(true)
+		setData(data => data.filter(entry => entry.latitude && entry.longitude)
+			.map(entry=> ({...entry, distance: getDistance(currentLocation, {latitude:entry.latitude || 0, longitude: entry.longitude || 0}) * 0.000621371 }))
```

Comment:
I don't think that `data` should be modified here; it should represent what comes from the API endpoint. The more appropriate place to do sorting and filtering computations is within the `sortAndFilterData` function that is called below.

Reply:
Also, we shouldn't sort by anything or show any distance if the location is not set.